polus-imaris-parser-plugin/src/main.py

```diff
@@ -0,0 +1,77 @@
+import argparse
+import pathlib
+import extract_ims_data
+import link_ims_ids
+import merge_ids_to_features
+import logging
+import time
+from pathlib import Path
+import os
+
+if __name__=="__main__":
+
+    #: Initialize the logger
+    logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                        datefmt='%d-%b-%y %H:%M:%S')
+    
+    logger = logging.getLogger("main")
+    logger.setLevel(logging.INFO)
+
+    #: Setup the argument parsing
+    logger.info("Parsing arguments...")
+    parser = argparse.ArgumentParser(prog='main', description='Parses metadata of Imaris files and outputs features in organized csv format.')
+    parser.add_argument('--inpdir', dest='inpdir', type=str,
+                        help='Input collection of ims files to be processed by this plugin', required=True)
+
+    parser.add_argument('--metaoutdir', dest='metaoutdir', type=str,
+                        help='Output metadata collection that will hold overall .xlsx file', required=True)
+
+    parser.add_argument('--outdir', dest='outdir', type=str,
+                        help='Output csv collection', required=True)
+
+    #: Parse the arguments
+    args = parser.parse_args()
+
+    #: Correct the filepath so that script accesses images stored in "metadata_files" directory instead of in "images" directory
+    inpdir = args.inpdir
+    logger.info("Old input directory: {}".format(inpdir))
+
+    if inpdir.endswith("/images") == True:
+        inpdir = inpdir[:-7]
+        inpdir = inpdir + "/metadata_files"
```

Comment:
Thank you for the suggestion. I have modified my code to check for subfolders named images then switch to that subfolder. From there it checks for metadata_files directory and switches to that one because that is where the .ims files get saved in wipp.

Reply:
Thank you. I have utilized the with_name functionality to switch to the appropriate directory and implemented better error handling if given incorrect input directories in my most recent commit

Reply:
Thanks, @Nicholas-Schaub! I have utilized the `with_name` functionality to switch to the appropriate `metadata_files` directory if it exists and implemented better error handling if given incorrect input directories in my most recent commit.